nni/retiarii/experiment/pytorch.py

```diff
@@ -131,7 +151,37 @@ def start(self, port: int = 8080, debug: bool = False) -> None:
         debug
             Whether to start in debug mode.
         """
-        super().start(port, debug)
+        atexit.register(self.stop)
+
+        self.id = management.generate_experiment_id()
+
+        if self.config.experiment_working_directory is not None:
```

Comment:
Is it a good idea to put these logics into Retiarii? Might as well create a new class in `nni.experiment`, called, for example, `_TunerCenteredExperiment`

Reply:
I'll consider this a temporary implementation and should be refactored in upcoming release.

Reply:
agree